clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -428,9 +429,28 @@ public void testHeartbeatState() {
         assertEquals(memberId, data.memberId());
         assertEquals(0, data.memberEpoch());
         assertNull(data.instanceId());
-        assertEquals(-1, data.rebalanceTimeoutMs());
+        assertEquals(DEFAULT_MAX_POLL_INTERVAL_MS, data.rebalanceTimeoutMs());
         assertEquals(Collections.singletonList(topic), data.subscribedTopicNames());
-        assertNull(data.serverAssignor());
+        assertEquals(ConsumerTestBuilder.DEFAULT_REMOTE_ASSIGNOR, data.serverAssignor());
+        assertNull(data.topicPartitions());
+        membershipManager.onHeartbeatRequestSent();
+        assertEquals(MemberState.JOINING, membershipManager.state());
+
+        // Explicitly change the subscription when gets fenced
+        String newTopic = "topic2";
+        when(membershipManager.state()).thenReturn(MemberState.FENCED);
+        subscriptions.subscribe(Collections.singleton(newTopic), Optional.empty());
```

Comment:
Hey @phong260702 . Sorry, I lost track of this PR. Feel free to ping me with my handle @lucasbru if that happens, I sometimes get lost in the GitHub noise.

I think the point of the comment was to have a case where use `transitionToFenced` but _not_ update the subscription. It seems you case does update the subscription.

Reply:
Oh I see. But isn't the test above with 'topic1' already done that?

Reply:
Looks like it's updating the subscription

```
        String topic = "topic1";
        subscriptions.subscribe(Collections.singleton(topic), Optional.empty());
        membershipManager.onSubscriptionUpdated();
        membershipManager.transitionToFenced(); // And indirect way of moving to JOINING state
```

Reply:
Oh I see. I;m a little bit confuse about the comment. I'll fix it.

Reply:
@lucasbru Should we use `transitionToFenced` as it will move the state to JOINING instead of FENCED

Reply:
yes. The ticket is about joining right? transitionTofenced is just a way to enter the rejoining state

Reply:
Got it. I've push a fix. Can you look at that.

Reply:
The test is still subscribing to a new topic

Reply:
Forgot that mybad. Fixed that.